fwprovider/access/resource_realm.go

```diff
@@ -0,0 +1,431 @@
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+
+	"github.com/bpg/terraform-provider-proxmox/fwprovider/config"
+	"github.com/bpg/terraform-provider-proxmox/proxmox"
+)
+
+var (
+	_ resource.Resource                     = (*realmResource)(nil)
+	_ resource.ResourceWithConfigure        = (*realmResource)(nil)
+	_ resource.ResourceWithImportState      = (*realmResource)(nil)
+	_ resource.ResourceWithConfigValidators = (*realmResource)(nil)
+)
+
+type realmResource struct {
+	client proxmox.Client
+}
+
+// NewRealmResource creates a new Realm resource.
+func NewRealmResource() resource.Resource {
+	return &realmResource{}
+}
+
+func (r *realmResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	cfg, ok := req.ProviderData.(config.Resource)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected config.Resource, got: %T", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = cfg.Client
+}
+
+func (r *realmResource) ConfigValidators(_ context.Context) []resource.ConfigValidator {
+	return []resource.ConfigValidator{
+		resourcevalidator.Conflicting(
+			path.MatchRoot("realm"),
+		),
+	}
+}
+
+func (r *realmResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Manages Realms on the Proxmox cluster",
+		MarkdownDescription: "Manages Realms on the Proxmox cluster.\n\n" +
+			"Realms are used for user authentication to a Proxmox node or cluster.\n",
+		Attributes: map[string]schema.Attribute{
+			"realm": schema.StringAttribute{
+				Description: "The ID of the realm which is of type string",
+				Required:    true,
+				Optional:    false,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"type": schema.StringAttribute{
+				Description: "Realm type",
+				Required:    true,
+				Optional:    false,
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(
+						regexp.MustCompile(`^ad$|^ldap$|^openid$|^pam$|^pve$`),
+						"must be one of [`ad`, `ldap`, `openid`, `pam`, `pve`]",
+					),
+				},
+			},
+			"acr_values": schema.StringAttribute{
+				Description: "Specifies the Authentication Context Class Reference values" +
+					"that theAuthorization Server is being requested to use for the Auth Request.",
+				Optional: true,
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(
+						regexp.MustCompile(`^[^\x00-\x1F\x7F <>#"]*$`),
+						"Not a valid ACR value",
+					),
+				},
+			},
+			"autocreate": schema.BoolAttribute{
+				Description: "Automatically create users if they do not exist.\n" +
+					"default is 0 or false",
```

Comment:
Sorry for being picky, but I have to call it out 😅 

"default is 0 or false" / "default is 1 or true" is applicable for Proxmox API, but seems confusing in the context of terraform resource. Also, we don't set any defaults for the attribute itself, so in fact this comment describes PVE behaviour, not the resource. I would rather say something like "Assumed `false` if not specified", or similar.

Reply:
You didn't specify default values for the attributes, and I think that's a right choice. We'd better to rely on PVE-supplied default, rather than hardcoding them in the provider. I have tons of issues with defaults in the legacy resources 😞